File: mobile/lib/services/exercises.service.dart
import '../models/exercise.dart';
import './base.service.dart';

class ExercisesService extends BaseService {
  ExercisesService() : super();

  Future<List<Exercise>> getExercises() async {
    final res = await request("/exercises", Method.get);
    if (res['exercises'] != null) {
      final exercises = (res['exercises'] as List<dynamic>)
          .map((exercise) => Exercise(exercise['id'], exercise['name'],
              exercise['mediaLink'], exercise['description']))
          .toList();
      return exercises;
    } else {
      throw "Failed to fetch exercises";
    }
  }
}

File: server/src/services/exercises.service.ts
import { query } from "../db/db";
import { Exercise } from "../models/Exercise";

class ExercisesService {
  async getExercises() {
    const { rows } = await query<string[]>(
      "SELECT id, name, media_link AS mediaLink, description FROM exercises"
    );
    const exercises = rows.map((row) => Exercise.fromDb(Object.values(row)));
    return exercises;
  }
}
export default ExercisesService;
